Microsoft.SCIM.WebHostSample/Provider/InMemoryUserProvider.cs

```diff
@@ -222,17 +203,20 @@ public override Task<Resource[]> QueryAsync(IQueryParameters parameters, string
 
                     } while (currentFilter.AdditionalFilter != null);
 
-                    results = results.Union(users.Select((Core2EnterpriseUser user) => user as Resource).ToList());
+                    predicate = predicate.Or(predicateAnd);
+
                 }
+
+                results = this.storage.Users.Values.Where(predicate.Compile());
             }
 
             if (parameters.PaginationParameters != null)
             {
                 int count = parameters.PaginationParameters.Count.HasValue ? parameters.PaginationParameters.Count.Value : 0;
-                return Task.FromResult(results.Take(count).ToArray());
+                return Task.FromResult((results.Take(count).ToArray()));
             }
             else
-                return Task.FromResult(results.ToArray());
+                return Task.FromResult((results.ToArray()));
```

Comment:
@CecilyK clean up the additional parentheses ;)

Reply:
Done :)